docs/standard/serialization/system-text-json/snippets/custom-contracts/SerializationCount.cs

```diff
@@ -0,0 +1,88 @@
+using System.Text.Json;
+using System.Text.Json.Serialization.Metadata;
+
+namespace Serialization
+{
+    // Custom attribute to annotate the property
+    // we want to be incremented.
+    [AttributeUsage(AttributeTargets.Property)]
+    class SerializationCountAttribute : Attribute
+    {
+    }
+
+    // Example type to serialize and deserialize.
+    class Product
+    {
+        public string Name { get; set; } = "";
+        [SerializationCount]
+        public int RoundTrips { get; set; }
+    }
+
+    public class SerializationCountExample
+    {
+        // Custom modifier that increments the value
+        // of a specific property on deserialization.
+        static void IncrementCounterModifier(JsonTypeInfo typeInfo)
+        {
+            foreach (JsonPropertyInfo propertyInfo in typeInfo.Properties)
+            {
+                if (propertyInfo.PropertyType != typeof(int))
+                    continue;
+
+                object[] serializationCountAttributes = propertyInfo.AttributeProvider?.GetCustomAttributes(typeof(SerializationCountAttribute), true) ?? Array.Empty<object>();
+                SerializationCountAttribute? attribute = serializationCountAttributes.Length == 1 ? (SerializationCountAttribute)serializationCountAttributes[0] : null;
+
+                if (attribute != null)
+                {
+                    Action<object, object?>? set = propertyInfo.Set;
+
+                    if (set != null)
+                    {
+                        propertyInfo.Set = (obj, value) =>
+                        {
+                            if (value != null)
+                            {
+                                // Increment the value by 1.
+                                value = (int)value + 1;
+                            }
+
+                            set(obj, value);
+                        };
+                    }
```

Comment:
```suggestion
                    Action<object, object?>? setProperty = propertyInfo.Set;
                    if (setProperty is not null)
                    {
                        propertyInfo.Set = (obj, value) =>
                        {
                            if (value != null)
                            {
                                // Increment the value by 1.
                                value = (int)value + 1;
                            }

                            setProperty (obj, value);
                        };
                    }
```

Avoid a variable name that is a contextual C# keyword.